ROCm 7 compatibility patch. 
Removes unused architectures from UCC-1.3.0 ROCm build.
Adds bool dependency.
author: Josh Maloney (University of Queensland)
diff -ruN ucc-1.3.0.bak/cuda_lt.sh ucc-1.3.0/cuda_lt.sh
--- ucc-1.3.0.bak/cuda_lt.sh	2024-04-18 01:27:30
+++ ucc-1.3.0/cuda_lt.sh	2026-02-21 12:13:47
@@ -28,7 +28,7 @@
 
 tmpcmd="${@:3}"
 if [[ "$tmpcmd" == *"amdclang"* ]]; then
-  cmd="${@:3:2} -x hip -target x86_64-unknown-linux-gnu --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx940 --offload-arch=gfx941 --offload-arch=gfx942 --offload-arch=gfx1030 --offload-arch=gfx1100 --offload-arch=gfx1101 --offload-arch=gfx1102 --offload-arch=native ${@:5} -fPIC -O3 -o ${pic_filepath}"
+  cmd="${@:3:2} -x hip -target x86_64-unknown-linux-gnu --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx942 --offload-arch=gfx1030 --offload-arch=gfx1100 --offload-arch=gfx1101 --offload-arch=gfx1102 --offload-arch=native ${@:5} -fPIC -O3 -o ${pic_filepath}"
 elif [[ "$tmpcmd" == *"hipcc"* ]]; then
   cmd="${@:3} -fPIC -o ${pic_filepath}"
 else
@@ -38,7 +38,7 @@
 $cmd
 
 if [[ "$tmpcmd" == *"amdclang"* ]]; then
-  cmd="${@:3:2} -x hip -target x86_64-unknown-linux-gnu --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx940 --offload-arch=gfx941 --offload-arch=gfx942 --offload-arch=gfx1030 --offload-arch=gfx1100 --offload-arch=gfx1101 --offload-arch=gfx1102 --offload-arch=native ${@:5} -O3 -o ${npic_filepath}"
+  cmd="${@:3:2} -x hip -target x86_64-unknown-linux-gnu --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx942 --offload-arch=gfx1030 --offload-arch=gfx1100 --offload-arch=gfx1101 --offload-arch=gfx1102 --offload-arch=native ${@:5} -O3 -o ${npic_filepath}"
 else
   cmd="${@:3} -o ${npic_filepath}"
 fi
Binary files ucc-1.3.0.bak/src/.DS_Store and ucc-1.3.0/src/.DS_Store differ
Binary files ucc-1.3.0.bak/src/components/.DS_Store and ucc-1.3.0/src/components/.DS_Store differ
Binary files ucc-1.3.0.bak/src/components/ec/.DS_Store and ucc-1.3.0/src/components/ec/.DS_Store differ
diff -ruN ucc-1.3.0.bak/src/components/ec/rocm/ec_rocm_executor_interruptible.c ucc-1.3.0/src/components/ec/rocm/ec_rocm_executor_interruptible.c
--- ucc-1.3.0.bak/src/components/ec/rocm/ec_rocm_executor_interruptible.c	2024-04-18 01:27:30
+++ ucc-1.3.0/src/components/ec/rocm/ec_rocm_executor_interruptible.c	2026-02-17 17:32:30
@@ -5,6 +5,7 @@
  * See file LICENSE for terms.
  */
 
+#include <stdbool.h>
 #include "ec_rocm_executor.h"
 #include "components/mc/ucc_mc.h"
 #include "components/ec/ucc_ec.h"
